File: WEEK21/이분탐색_입국심사.py
def solution(n, times):
    answer = 0
    
    left, right = 1, n * max(times)
    
    while left <= right:
        mid = (left + right) // 2
        total = 0
        
        for time in times:
            total += mid // time
        
        if total >= n:
            answer = mid
            right = mid - 1
        elif total < n:
            left = mid + 1
    
    return answer

File: WEEK21/이분탐색_입국심사.js
function solution(n, times) {
  var answer = 0;
  let left = 1;
  let right = n * Math.max(...times);

  while (left <= right) {
    let mid = parseInt((left + right) / 2);
    let total = 0;

    times.forEach((time) => {
      total += parseInt(mid / time);
    });

    if (total >= n) {
      answer = mid;
      right = mid - 1;
    } else if (total < n) {
      left = mid + 1;
    }
  }

  return answer;
}
